src/algorithms/cDIIS.jl

```diff
@@ -0,0 +1,169 @@
+#
+# Direct inversion in the iterative subspace
+#
+
+"""
+    Container type for the state of one spin type for cDIIS
+"""
+mutable struct cDIISstate
+    fock::CircularBuffer
+    error::CircularBuffer
+    errorOverlaps::CircularBuffer
+end
+
+"""
+    cDIIS
+"""
+mutable struct cDIIS <: Accelerator
+    n_prev_steps::Int
+    stateα::cDIISstate
+    stateβ::cDIISstate
+
+    function cDIIS(problem::ScfProblem; n_prev_steps = 5, kwargs...)
+        stateα = cDIISstate(CircularBuffer{AbstractArray}(n_prev_steps),
+            CircularBuffer{AbstractArray}(n_prev_steps),
+            CircularBuffer{AbstractArray}(sum(1:n_prev_steps -1))
+        )
+        stateβ = deepcopy(stateα)
+        new(n_prev_steps, stateα, stateβ)
+    end
+end
+
+"""
+    Helper function.
+    pushes current fock and error matrices to states of both spin types
+"""
+function pushIterateToState!(iterate::ScfIterState, acc::cDIIS)
+    pushfirst!(acc.stateα.fock, iterate.fock[:,:,1])
+    pushfirst!(acc.stateα.error, iterate.error[:,:,1])
+    if size(iterate.fock, 3) == 2
+        pushfirst!(acc.stateβ.fock, iterate.fock[:,:,2])
+        pushfirst!(acc.stateβ.error, iterate.error[:,:,2])
+    else
+        acc.stateβ = acc.stateα
```

Comment:
I am not too happy that `stateβ` is dragged along even if no `β`-block is present in error or fock.